nengo_spa/algebras/vtb_algebra.py

```diff
@@ -155,6 +218,81 @@ def invert(self, v, sidedness=ElementSidedness.TWO_SIDED):
         sub_d = self._get_sub_d(len(v))
         return v.reshape((sub_d, sub_d)).T.flatten()
 
+    def binding_power(self, v, exponent):
+        r"""Returns the binding power of *v* using the *exponent*.
+
+        The binding power is defined as binding (*exponent*-1) times bindings
+        of *v* to itself.
+
+        Fractional binding powers are supported for "positive" vectors if SciPy
+        is available.
+
+        Note the following special exponents:
+
+        * an exponent of -1 will return the inverse,
+        * an exponent of 0 will return the identity vector,
+        * and an *exponent* of w1cne will return *v* itself.
+
+        Be aware that the binding power for the VTB algebra does *not* satisfy
+        the usual properties of exponentiation:
+
+        * :math:`\mathcal{B}(v^a, v^b) = v^{a+b}` does *not* hold,
+        * :math:`(v^a)^b = v^{ab}` does *not* hold.
+
+        Parameters
+        ----------
+        v : (d,) ndarray
+            Vector to bind repeatedly to itself.
+        exponent : int or float
+            Exponent of the binding power.
+
+        Returns
+        -------
+        (d,) ndarray
+            Binding power of *v*.
+
+        See also
+        --------
+        .sign
+        """
+
+        try:
+            from scipy.linalg import fractional_matrix_power
+        except ImportError:
+            if int(exponent) == exponent:
+                exponent = int(exponent)
+                # Provide fallback for integer-only powers
+
+                def fractional_matrix_power(m, exp):
+                    power = np.eye(len(m))
+                    for _ in range(exp):
+                        power = np.dot(power, m)
+                    return power
+
+            else:
+                raise ValueError(
+                    "Fractional VTB binding powers require SciPy to be available."
+                )
+
+        if exponent == 0:
+            return self.identity_element(len(v), sidedness=ElementSidedness.RIGHT)
+
+        if int(exponent) != exponent and not self.sign(v).is_positive():
+            raise ValueError(
+                "Fractional binding powers are only supported for 'positive' vectors."
+            )
+
+        if exponent < 0:
+            v = self.invert(v, sidedness=ElementSidedness.RIGHT)
+
+        sub_d = self._get_sub_d(len(v))
+        power = fractional_matrix_power(
+            v.reshape((sub_d, sub_d)) * np.sqrt(sub_d), abs(exponent) - 1
```

Comment:
Would there be any possibility for problem here if `0 < abs(exponent) < 1`?

Reply:
* [ ] If my math is correct that should not be a problem, but I will double check.